File: HW10/hw10/app/src/main/java/com/example/hw10/MainActivity.java
package com.example.hw10;

import androidx.appcompat.app.AppCompatActivity;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.os.Handler;
import android.view.View;
import android.widget.ProgressBar;

public class MainActivity extends AppCompatActivity {

    static int progress;
    ProgressBar progressBar;
    int progressStatus = 0;
    Handler handler = new Handler();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        progress = 0;
        progressBar = (ProgressBar) findViewById(R.id.progressbar);

        IntentFilter filter = new IntentFilter("progress_update");
        LocalBroadcastManager.getInstance(this).registerReceiver(progressReceiver, filter);

    }

    public void startService(View view) {
        startService(new Intent(getBaseContext(), MyService.class));
    }

    public void stopService(View view) {
        stopService(new Intent(getBaseContext(), MyService.class));
    }


    private BroadcastReceiver progressReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            if ("progress_update".equals(intent.getAction())) {
                int progress = intent.getIntExtra("progress", 0);
                // Update ProgressBar with the received progress value
                progressBar.setProgress(progress);
            }
        }
    };
}

File: HW8/hw8/app/src/main/java/com/example/hw8/MainActivity.kt
package com.example.hw8

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.ListView
import androidx.activity.result.ActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.core.content.ContentProviderCompat.requireContext
import androidx.lifecycle.lifecycleScope
import androidx.room.Room
import com.google.android.material.floatingactionbutton.FloatingActionButton
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {

    private var contactsList: List<Contact> = emptyList()
    private lateinit var fabAdd: FloatingActionButton
    lateinit var listView: ListView
    lateinit var adapter: ContactAdapter

    private lateinit var contactDao: ContactDao

    val startForResult =
        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
            if (result.resultCode == Activity.RESULT_OK) {
                val results = result.data?.extras


                lifecycleScope.launch {
                    contactDao.insert(
                        Contact(
                            name = results?.get("name").toString(),
                            phone = results?.get("phone").toString(),
                            email = results?.get("email").toString()
                        )
                    )
                    contactsList = contactDao.getAllContacts()
                }
                adapter = ContactAdapter(this, contactsList)
                listView.adapter = adapter

                adapter.notifyDataSetChanged()


//                textViewName.text = "Name: ${results?.get("name")}"
//                textViewPhone.text = "Phone: ${results?.get("phone")}"
//                textViewEmail.text = "Email: ${results?.get("email")}"


            }
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        fabAdd = findViewById(R.id.fabAddContact)
        listView = findViewById(R.id.listView)

        adapter = ContactAdapter(this, contactsList)
        listView.adapter = adapter



        fabAdd.setOnClickListener {
            startForResult.launch(Intent(this, ActivityTwo::class.java))
        }

        listView.setOnItemClickListener { adapterView, view, pos, l ->
            val selectedItem = contactsList[pos]
            val intent = Intent(this, ActivityThree::class.java)
            intent.putExtra("name", selectedItem.name)
            intent.putExtra("phone", selectedItem.phone)
            intent.putExtra("email", selectedItem.email)
            startActivity(intent)
        }

        val db = Room.databaseBuilder(
            applicationContext,
            ContactsDatabase::class.java, "contactsdatabase"
        ).build()

        contactDao = db.contactDao()

        lifecycleScope.launch {

            contactsList = contactDao.getAllContacts()
            adapter = ContactAdapter(this@MainActivity, contactsList)
            listView.adapter = adapter
            adapter.notifyDataSetChanged()

        }
    }
}